src/main/java/strategy/PurchaseOperation.java

```diff
@@ -0,0 +1,10 @@
+package strategy;
+
+import model.FruitTransaction;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public int getQuantityToCalculate(FruitTransaction fruitTransaction) {
+        return -fruitTransaction.getQuantity();
```

Comment:
Handling Purchase operation. Check result balance before saving it in the Storage - it should be positive. Throw `RuntimeException` in case the balance is negative.